torch/csrc/api/include/torch/nn/functional/activation.h

```diff
@@ -40,6 +40,25 @@ inline Tensor logsigmoid(const Tensor& input) {
   return torch::log_sigmoid(input);
 }
 
+inline Tensor softmax(const Tensor& input, const SoftmaxOptions& options) {
+  int dim = options.dim();
+  torch::Dtype dtype = options.dtype();
+
+  if (dim == -1) {
+    int input_dim = input.dim();
+    if (input_dim == 0 || input_dim == 1 || input_dim == 3) {
```

Comment:
Please extract this to _get_softmax_dim(...) as it is in [python](https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1166) and reuse it in LogSoftmax. Use [TORCH_WARN](https://github.com/pytorch/pytorch/blob/master/c10/util/Exception.h#L287) for the [warning](https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1168).

Reply:
I fixed it.